conda_build/build.py

```diff
@@ -582,7 +582,7 @@ def test(m, verbose=True, channel_urls=(), override_channels=False):
         else:
             test_file = join(tmp_dir, 'run_test.sh')
             # TODO: Run the test/commands here instead of in run_test.py
-            cmd = ['/bin/bash', '-x', '-e', test_file]
+            cmd = ['/bin/sh', '-x', '-e', test_file]
```

Comment:
> What if there are bashisms in the build.sh script of a recipe? We've never made the promise to work with other shells.

Then I guess we should make all the scripts work with `sh`, as it's the only reliable shell that one can find at a fixed location on any POSIX compliant OS.
To be honest, let's not over-engineer this change, it's a 0 cost issue. I did not found any script in conda-recipes that rely on bash-specific syntax, it's not like build scripts are very complex, most of them just run some very basic commands, there's no pragmatic need for `bash` here.

> I assume BSDs have a package for bash, so why not use the path provided to bash for each BSD instead of changing the default shell?

Because it's not just a BSD issue here. There is no guarantee even on linux that bash will be installed in `/bin`.

Also note that if a build script _really_ needs bash, then it should call it itself with `/usr/bin/env bash`, but no need to force bash for everyone, when really all these scripts do is run some commands.